src/portable/linux.rs

```diff
@@ -103,6 +103,7 @@ Documentation=https://edgedb.com/
 
 [Socket]
 ListenStream=127.0.0.1:{port}
+ListenStream=[::1]:{port}
```

Comment:
What would happen if the system has no support for IPv6?  Should we listen on `localhost:{port}` instead?

Reply:
I've been testing this, looks like the service will fail to start in that case. `localhost` is not always good because:

1. Some distribution don't have `::1 localhost` in `/etc/hosts` - my Ubuntu Jammy has only `::1     ip6-localhost ip6-loopback`
2. GitHub Action has `::1 localhost` but it doesn't support IPv6 if I remember correctly

Reply:
Can't we have a second Socket unit for "ListenStream=[::1]:{port}" not `WantedBy` by the service? Which would presumably make it an optional socket to start with?

Reply:
I think we can try listening on `[::1]:0` in the CLI and then create an appropriate unit?

Reply:
Oh, there's a better way.

```
ListenStream={port}
```

This way systemd will automatically listen on all available IP versions.

Reply:
Nevermind, this will also listen on all interfaces, which is not what we want.

Reply:
> Can't we have a second Socket unit for "ListenStream=[::1]:{port}" not `WantedBy` by the service? Which would presumably make it an optional socket to start with?

Even if that trick will work on linux, I think there is no such feature on macos.

Reply:
AFAIK non-working IPv6 on macOS is probably a lot less likely, so that's fine.

Reply:
> I think we can try listening on `[::1]:0` in the CLI and then create an appropriate unit?

This seems like the most reliable approach to me.